src/main/java/dibo/Parser.java

```diff
@@ -0,0 +1,146 @@
+                throw new DiboException("Oh no sir! Please state the deadline of the task :D");
+            } catch (DateTimeParseException e) {
+                throw new DiboException("Oh no sir! Please state the deadline of the task "
+                        + "in this format: yyyy-mm-dd");
+            }
+        case "event":
+            if (!hasDescription(fullCommand)) {
+                throw new DiboException("Oh no sir! We need a description for your task. "
+                        + "This will enable us to better keep track of your tasks.");
+            }
+            try {
+                String descriptionEvent = getDescription(fullCommand, "event");
+                LocalDate startDate = getStartDate(fullCommand);
+                LocalDate endDate = getEndDate(fullCommand);
+                if (!startDate.isBefore(endDate)) {
+                    throw new DiboException("Oh no sir! Your start date must be before your end date :(");
+                }
+                return new AddEventCommand(descriptionEvent, startDate, endDate);
+            } catch (IndexOutOfBoundsException e) {
+                throw new DiboException("Oh no sir! Please state the start and end of the task :D");
+            } catch (DateTimeParseException e) {
+                throw new DiboException("Oh no sir! Please state the deadline of the task "
+                        + "in this format: yyyy-mm-dd");
+            }
+        default:
+            throw new DiboException("Oh no sir! There is no such task type :(");
+        }
+    }
+
+    private static boolean hasDescription(String fullCommand) {
+        String[] details = fullCommand.split("/");
+        String[] typeAndDescription = details[0].split(" ");
+        return typeAndDescription.length == 2;
+    }
+
+    private static String getDescription(String fullCommand, String type) {
+        String[] details = fullCommand.split("/");
+        return details[0].substring(type.length()).trim();
+    }
+
+    private static LocalDate getByDate(String fullCommand)
+            throws DateTimeParseException, IndexOutOfBoundsException {
+        String[] details = fullCommand.split("/");
+        String byString = details[1].substring(2).trim();
+        return LocalDate.parse(byString, Parser.INPUT_FORMAT);
+    }
+    private static LocalDate getStartDate(String fullCommand)
+            throws DateTimeParseException, IndexOutOfBoundsException {
+        String[] details = fullCommand.split("/");
+        String startString = details[1].substring(4).trim();
+        return LocalDate.parse(startString, Parser.INPUT_FORMAT);
+    }
+
+    private static LocalDate getEndDate(String fullCommand)
+            throws DateTimeParseException, IndexOutOfBoundsException {
+        String[] details = fullCommand.split("/");
+        String endString = details[2].substring(2).trim();
+        return LocalDate.parse(endString, Parser.INPUT_FORMAT);
+    }
```

Comment:
Not sure I like how these helper methods are `private static` rather than just `private`. Is there a rationale behind this?

Reply:
Actually, I didn't take notice of this before. I will take note in the future! Thank you so much 😄